compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -229,7 +229,7 @@ enum ImplTraitContext<'b, 'a> {
         /// them. This includes lifetimes bound since we entered this context.
         /// For example:
         ///
-        /// ```
+        /// ```ignore HELP (how would you make a stub trait for this?)
         /// type A<'b> = impl for<'a> Trait<'a, Out = impl Sized + 'a>;
```

Comment:
Should we ignore this one as just an "illustration" then? The original single line of code gets quite buried when we add all that. If we figure out how to make it compile we could just hide the extra lines, but it would still be harder to read the un-rendered source (is that at all a priority for us? I've been considering it moderately important as my vscode doesn't have built-in doc rendering unless I hover on an item which is tedious and doesn't work for everything, but if we expect most future readers to have rendering then I can see why we wouldn't bother with it.)

Reply:
I think it's semi important to figure out what the comment is describing in the first place - it's unclear to me now even having read it several times and played around with the example, so it will almost certainly be unclear to whoever reads it next without that context. But I agree making it compile is probably not worth it.

Reply:
It's unclear to me, too. Igot all the way to

```rust
#![feature(type_alias_impl_trait)]
trait Trait<'a> { type Out: 'a;}
impl<'a> Trait<'a> for i32 { type Out = String;}
type A = impl for<'a> Trait<'a, Out = impl Sized + 'a>;
fn foo() -> A {
    0_i32
}

```

which is just a bug that it ain't accepted.i'll openable issuefor it And try to figure out the details of what's going on with this field at the same time.